### ISSUE TITLE
Default hasher doesn't work in wasm

### ISSUE THREAD

**Issue by chinedufn** [alexheretic/glyph-brush#61]
`twox-hash` doesn't work in `.wasm`, I believe because it's allowed `rand` versions include versions below `rand = 0.6.0` (when rand first landed `.wasm` support I believe).

So my app crashes if I try and use `glyph_brush` right now.

To get around this, I thought of using a different hasher:

```rust
let mut brush = GlyphBrushBuilder::using_font_bytes(font)
.section_hasher(RandomState::default())
.build();
```

No dice .. `twox-hash` seems to still get called

![image](https://user-images.githubusercontent.com/2099811/55924170-b2a81680-5bd6-11e9-81e4-e5f177b93c8d.png)

Anything come to mind as to why?

In short - I'd like to prevent `twox-hash` from getting called in any way.

Thanks!

**Comment by chinedufn** [comment]
Alright so it looks like `using_font_bytes` ends up hitting this line

https://github.com/alexheretic/glyph-brush/blob/1ef9fdef25c73ee80559fba231b0b16a4af4f967/glyph-brush/src/glyph_brush/builder.rs#L65

Which uses the DefaultSectionHasher (twox-hash). I'll check the API docs to see if I can set my hasher being adding fonts

**Comment by chinedufn** [comment]
Alright looked over the `GlyphBrushBuilder` and `GlyphBrush` docs and it doesn't appear to be possible to create a `GlyphBrush` without hitting twox-hash right now.

Is this true - or am I missing something?

Cheers!

**Comment by alexheretic** [comment]
Thanks for reporting this, twox-hash supports a range of rand versions including 0.6, so maybe try `cargo update`. According to cargo tree my desktop usage has rand 0.6.

```
│   │   └── twox-hash v1.1.2
│   │       └── rand v0.6.5
```

I'll fix creating the default hasher as it will avoid a random number gen too in the cases you're switching it out.

I'll have to look into adding a wasm32 CI.

**Comment by alexheretic** [comment]
Actually I just opted to work around this issue as the generics of the builder are awkward to change without breaking downstream crates.

```rust
#[cfg(target_arch = "wasm32")]
pub type DefaultSectionHasher = std::hash::BuildHasherDefault<twox_hash::XxHash>;
```
This should avoid any rand usage as it builds the `XxHash` with a zero seed.

@chinedufn can you see if this fixes the issue for you?

**Comment by chinedufn** [comment]
@alexheretic I'm all set - thanks!

> I'll have to look into adding a wasm32 CI.

This would be awesome!

---

As always - thanks for the super quick help!
